src/components/comment-form/comment-form.tsx

```diff
@@ -1,21 +1,59 @@
-import React, {useState} from 'react';
+import React, { useState, FormEvent, ChangeEvent } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { postCommentAction } from '../../store/api-actions';
+import { selectCurrentOffer } from '../../store/selectors';
+import { CommentToSend } from '../../types/comments';
+import { AppRoute } from '../../const';
+import { fetchCommentsAction } from '../../store/api-actions';
 
 export const CommentForm = () => {
+  const currentOffer = useAppSelector(selectCurrentOffer);
   const [formData, setFormData] = useState({
-    rating: '',
-    comment: ''
-  });
+    comment: '',
+    rating: 0,
+    id: currentOffer?.id
+  } as CommentToSend);
 
-  const onInputChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
-    const {name, value} = event.target;
-    setFormData({
-      ...formData,
-      [name]: value
-    });
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const inputStarsCountChangeHandler = (evt: ChangeEvent<HTMLInputElement>) => {
+    const value = Number(evt.target.value);
+    if (currentOffer) {
+      setFormData({
+        ...formData,
+        rating: value
+      });
+    }
+  };
+
+  const textareaChangeHandler = (evt: ChangeEvent<HTMLTextAreaElement>) => {
+    const {value} = evt.target;
+    if (currentOffer) {
+      setFormData({
+        ...formData,
+        comment: value
+      });
+    }
+  };
+
+  const formSubmitHandler = (evt: FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    dispatch(postCommentAction(formData));
+    if (currentOffer) {
+      dispatch(fetchCommentsAction(currentOffer));
+      navigate(`${AppRoute.Offer}${currentOffer.id}`);
+      setFormData({
+        ...formData,
+        comment: '',
+        rating: 0
+      });
+    }
```

Comment:
Есть альтернативный подход стилизовать код, это когда делаешь возврат сначала, а потом логику. Так уменьшается вложенность кода. Но это вкусовщина, поэтому менять или нет - на твой выбор

```suggestion
    if (!currentOffer) return;
    
    dispatch(fetchCommentsAction(currentOffer));
    navigate(`${AppRoute.Offer}${currentOffer.id}`);
    setFormData({
      ...formData,
      comment: '',
      rating: 0
    });
```